### PR TITLE
Remove Deprecated Modules

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by harendra-kumar** [comment]
We should remove only after at least two major releases of deprecation that means it has to be deprecated in 0.7 and 0.8, then we can remove in 0.9.

We can remove Streamly.Internal.Data.Time module for example because it remained deprecated in 0.7 and 0.8.

**Comment by harendra-kumar** [comment]
As I said earlier, we should not remove too early. We should remove only if the modules/functions were deprecated in 0.7 or earlier. Many of the functions removed in this PR were deprecated only in 0.8.

File: src/Streamly/Internal/Data/Stream/IsStream/Transform.hs [hunk 1]
```diff
@@ -705,29 +702,6 @@ postscan fld = fromStreamD . D.postscanOnce fld . toStreamD
...
 
--- XXX It may be useful to have a version of scan where we can keep the
--- accumulator independent of the value emitted. So that we do not necessarily
--- have to keep a value in the accumulator which we are not using. We can pass
--- an extraction function that will take the accumulator and the current value
--- of the element and emit the next value in the stream. That will also make it
--- possible to modify the accumulator after using it. In fact, the step function
--- can return new accumulator and the value to be emitted. The signature would
--- be more like mapAccumL. Or we can change the signature of scanx to
--- accommodate this.
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/IsStream/Transform.hs:716]: We can keep this TODO comment.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
I do not see Streamly or Streamly.Data.Unicode.Stream or Streamly.Memory.Array being deprecated in 0.7.0.

**Comment by harendra-kumar** [comment]
Please rebase to resolve the conflicts.

File: src/Streamly/Memory/Array.hs [hunk 2]
```diff
@@ -71,4 +71,4 @@ module Streamly.Memory.Array
     )
 where
 
-import Streamly.Internal.Data.Array.Foreign as A
+import Streamly.Internal.Data.Array.Foreign as A
```
Comment (harendra-kumar) [inline src/Streamly/Memory/Array.hs:74]: This file should not have any change, should be reverted.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Comment by harendra-kumar** [comment]
Please squash the changes.

**Comment by harendra-kumar** [comment]
Can we add a changelog entry, saying removal of function deprecated in or before 0.7.0?

**Comment by rnjtranjan** [comment]
Fixes: #1477

File: docs/Changelog.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 <!-- See rendered changelog at https://streamly.composewell.com -->
+## 0.9.0 (Mar 2022)
```
Comment (harendra-kumar) [inline docs/Changelog.md:4]: We use "Unreleased" here instead of 0.9.0 .

**Comment by harendra-kumar** [comment]
It seems many are still running 0.7.x, so let's remove only those that were deprecated in or before 0.6 .

**Comment by rnjtranjan** [comment]
opened another PR : https://github.com/composewell/streamly/pull/1552

**Pull request closed without merging**